docs/guide/component-testing.md

```diff
@@ -0,0 +1,402 @@
+---
+title: Component Testing | Guide
+outline: deep
+---
+
+# Component Testing
+
+Vitest provides comprehensive support for component testing across multiple frameworks including Vue, React, Svelte, Lit, Marko, and more. Component testing allows you to test your UI components in isolation, ensuring they render correctly, handle user interactions, and maintain expected behavior.
+
+## Overview
+
+Component testing in Vitest can be done in two ways:
+
+1. **Browser Mode** (Recommended) - Run tests in a real browser environment
+2. **Node.js with DOM simulation** - Using jsdom or happy-dom for faster execution
+
+For the most accurate testing experience, we recommend using [Browser Mode](/guide/browser/) as it provides a real browser environment where your components actually run.
+
+## Browser Mode Component Testing
+
+Browser Mode is the recommended approach for component testing as it provides the most accurate testing environment. Your tests run in real browsers using providers like Playwright, WebdriverIO, or preview mode.
+
+### Setup
+
+The easiest way to get started with browser mode is using Vitest's init command:
+
+```bash
+vitest init browser
+```
+
+This will automatically install the necessary dependencies and configure your project for browser testing.
+
+Alternatively, you can set it up manually:
+
+::: code-group
+```bash [npm]
+npm install --save-dev @vitest/browser playwright
+```
+
+```bash [yarn]
+yarn add --dev @vitest/browser playwright
+```
+
+```bash [pnpm]
+pnpm add -D @vitest/browser playwright
+```
+:::
+
+Then configure browser mode in your `vitest.config.ts`:
+
+```ts
+import { defineConfig } from 'vitest/config'
+
+export default defineConfig({
+  test: {
+    browser: {
+      enabled: true,
+      name: 'chromium',
+      provider: 'playwright',
```

Comment:
That is no longer how provider is specified in Vitest 4. See: https://main.vitest.dev/guide/browser/playwright.html